types/wlr_foreign_toplevel_management_v1.c

```diff
@@ -127,13 +130,17 @@ static void foreign_toplevel_handle_close(struct wl_client *client,
 }
 
 static void foreign_toplevel_handle_set_rectangle(struct wl_client *client,
-		struct wl_resource *resource, struct wl_resource *surface,
+		struct wl_resource *resource, struct wl_resource *surface_resource,
 		int32_t x, int32_t y, int32_t width, int32_t height) {
 	struct wlr_foreign_toplevel_handle_v1 *toplevel =
 		toplevel_handle_from_resource(resource);
 	if (!toplevel) {
 		return;
 	}
+	struct wlr_surface *surface = wlr_surface_from_resource(surface_resource);
+	if (!surface) {
```

Comment:
I don't think a surface can be inert.

Reply:
Looks like you're right. I skimmed all the `wl_resource_set_user_data()` calls in wlroots but didn't find any setting the `wl_surface` resource's user data to `NULL`.

Reply:
#2597